### PR TITLE
test: include 'whiteboard' in preserved parameters [V2]

### PR DESCRIPTION
**Pull request by apahim**
v2:
- Include unit test.

v1: #2423

### CODE REVIEW

File: selftests/unit/test_runner_queue.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+from multiprocessing import queues
+
+from avocado.core.job import Job
+from avocado.core.result import Result
+from avocado.core.runner import TestRunner
+from avocado.core.tree import TreeNode
+
+
+class TestRunnerQueue(unittest.TestCase):
```
Comment (ldoktor) [inline selftests/unit/test_runner_queue.py]: @apahim, how about simply:

```
>>> import avocado
>>> q=avocado.core.job.Job()
>>> a=avocado.core.test.DryRunTest(job=q)
>>> a.whiteboard = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
>>> a.get_state()
{'status': None, 'whiteboard': 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa', 'time_start': -1, 'name': '0-DryRunTest', 'job_unique_id': 'f8941aac98cadc2df2f95b2b3ec647e4fa6eea32', 'class_name': 'DryRunTest', 'traceback': None, 'time_elapsed': -1, 'time_end': -1, 'paused': False, 'running': False, 'fail_class': None, 'params': <AvocadoParams *: * ([]),0: /test/* ([])>, 'logdir': '/home/medic/avocado/job-results/job-2018-02-06T12.20-80b9c78/test-results/0-DryRunTest', 'timeout': None, 'fail_reason': None, 'logfile': '/home/medic/avocado/job-results/job-2018-02-06T12.20-80b9c78/test-results/0-DryRunTest/debug.log', 'job_logdir': '/home/medic/avocado/job-results/job-2018-02-06T12.20-f8941aa'}
```

(obviously with tmpdir set in job).

Comment (apahim) [reply selftests/unit/test_runner_queue.py]: I'm testing the queue content from the runner perspective.

**Comment by apahim** [comment]
The good reason is to test the queue content. That's where the whiteboard has to exist, regardless the test implementation to put the information in the queue.

**Comment by ldoktor** [comment]
Well I did not really nack it, I just asked for more details (github does not have the "need_info" flags so I used this instead so it list of PRs I can easily see whether it's responded or no). Anyway I still think it make sense to ask those kind of questions because unit tests are suppose to be minimal and should ideally check the only classes they intend to test. Here it looked to me unnecessary complex because we have other tests focused on exercising the queue and the TestStatus class itself.

My point was that this patch is focused on having `whiteboard` included in `get_state`, which can be checked easily without other internal deps. In case we change, while this selftest uses many core APIs to check basically the same thing but changing anything used there but not-dependant-to-get-state will require change of this test which is the opposite of unit-testing.

Anyway we do have coding-style-guides and I think we still need to define what should and should not go in different Avocado components (avocado.core, avocado.utils, optional_plugins, selftests.functional, selftest.unit, examples).

Also I discussed the selftests issue with colleagues from kernel/systemd and they said even in unittests external/our_optional deps are fine if that makes the result possible or significantly cleaner/faster. I'm still inclined to avoiding it as much as possible there, but after arguing with them I'm not that strongly convinced.

**Comment by clebergnu** [comment]
@ldoktor unfortunately I can not read your "Request Changes" on GitHub as a request for "more info", when you:
 * Have been using GitHub for age
 * Know you can *comment* on a PR
 * Constantly request different implementations on your reviews

At the very least, you're misusing the tool and causing miscommunication.  The other part of the conversation doesn't belong in this PR.

**Pull request merged**